inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/deformable_convolution_gpu_bfyx_ref.cl

```diff
@@ -48,26 +51,33 @@ KERNEL(deformable_convolution_gpu_bfyx_ref)(
     for (uint dg = 0; dg < DEFORMABLE_GROUPS; ++dg)
     {
         const int ifm_offset = dg * channel_per_deformable_group;
-        const int trans_offset = b * INPUT1_BATCH_PITCH +
-                                 dg * 2 * FILTER_SIZE_Y * FILTER_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_X;
+        const int dg_offset = dg * FILTER_SIZE_Y * FILTER_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_X;
+        const int trans_offset = b * INPUT1_BATCH_PITCH + 2 * dg_offset;
+#if DEFORMABLE_MASK_ENABLED
+        const int mask_offset = b * INPUT2_BATCH_PITCH + dg_offset;
+#endif
         for (uint j = 0; j < FILTER_SIZE_Y ; ++j)
         {
             const int input_offset_y = input_y + j * DILATION_SIZE_Y;
             for (uint i = 0; i < FILTER_SIZE_X ; ++i)
             {
                 const int trans_y_idx = ((2 * (j * FILTER_SIZE_X + i)) * OUTPUT_SIZE_Y + y) * OUTPUT_SIZE_X + x;
                 float transformed_y = input_offset_y + (float)trans[trans_offset + trans_y_idx];
-
-                const bool zero_y = transformed_y >= INPUT0_SIZE_Y || transformed_y < 0;
-
                 const int input_offset_x = input_x + i * DILATION_SIZE_X;
-
                 const int trans_x_idx = ((2 * (j * FILTER_SIZE_X + i) + 1) * OUTPUT_SIZE_Y + y) * OUTPUT_SIZE_X + x;
                 float transformed_x = input_offset_x + (float)trans[trans_offset + trans_x_idx];
-
-                const bool zero_x = transformed_x >= INPUT0_SIZE_X || transformed_x < 0;
-                if(!zero_y) {
-                    if(!zero_x) {
+#if BILINEAR_INTERPOLATION_PAD
+                const bool x_is_out_of_boundaries = (int)transformed_x >= INPUT0_SIZE_X || (int)transformed_x <= -1;
+                const bool y_is_out_of_boundaries = (int)transformed_y >= INPUT0_SIZE_Y || (int)transformed_y <= -1;
+#else
+                const bool x_is_out_of_boundaries = transformed_x >= INPUT0_SIZE_X || transformed_x < 0;
+                const bool y_is_out_of_boundaries = transformed_y >= INPUT0_SIZE_Y || transformed_y < 0;
+#endif
+                if(!y_is_out_of_boundaries) {
+                    if(!x_is_out_of_boundaries) {
```

Comment:
Maybe it makes sence to combine both conditions into one with "&"?

Reply:
done